### PR TITLE
MRG Round eeglab latencies

### LINKED ISSUE #4954: dealing with eeglab float event latencies
Eeglab represents event latencies with floats after downsampling (to suggest that events are no longer precisely co-occuring with specific time samples). Instead of turning these to int representation, which truncates the value, we should be rounding.

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
Redo of #5377

Closes #4954

### CODE REVIEW

**Comment by agramfort** [comment]
@jona-sassenhagen can you see why travis complains?

@mmagnuski ok with you?

**Comment by jona-sassenhagen** [comment]
Back to WIP :(

File: mne/io/eeglab/tests/test_eeglab.py [hunk 1]
```diff
@@ -107,7 +107,7 @@ def test_io_set_raw(fnames, tmpdir):
     for event in eeg.event:  # old version allows integer events
         event.type = 1
     assert_equal(read_events_eeglab(eeg)[-1, -1], 1)
-    eeg.event = eeg.event[0]  # single event
+    eeg.event = eeg.event[0] -.1  # single event
```
Comment (mmagnuski) [inline mne/io/eeglab/tests/test_eeglab.py]: Why `-.1` and not `-1` (I assume this is matlab to python index conversion)?

Comment (mmagnuski) [reply mne/io/eeglab/tests/test_eeglab.py]: but -1 was already corrected for elsewhere (earlier), no?

Comment (mmagnuski) [reply mne/io/eeglab/tests/test_eeglab.py]: oh, these are tests :)

File: mne/io/eeglab/tests/test_eeglab.py [hunk 2]
```diff
@@ -141,8 +142,11 @@ def test_io_set_raw(fnames, tmpdir):
     shutil.copyfile(op.join(base_dir, 'test_raw.fdt'),
                     negative_latency_fname.replace('.set', '.fdt'))
     event_id = {eeg.event[0].type: 1}
-    pytest.raises(ValueError, read_raw_eeglab, montage=montage, preload=True,
-                  event_id=event_id, input_fname=negative_latency_fname)
```
Comment (jona-sassenhagen) [inline mne/io/eeglab/tests/test_eeglab.py]: @larsoner this change as intended?

Comment (jona-sassenhagen) [reply mne/io/eeglab/tests/test_eeglab.py]: I mean, this line *should* be a warning now, no longer an error, right?

File: mne/io/eeglab/eeglab.py [hunk 3]
```diff
@@ -675,9 +675,12 @@ def read_events_eeglab(eeg, event_id=None, event_id_func='strip_to_integer',
         logger.info('No events found, returning empty stim channel ...')
         return np.zeros((0, 3))
 
-    if (latencies < 0).any():
+    if (latencies < -1).any():
         raise ValueError('At least one event sample index is negative. Please'
                          ' check if EEG.event.sample values are correct.')
+    if (latencies == -1).any():
+        warn("Some events have a sample index of -1. This may cause issues "
+             "during, e.g., resampling.")
```
Comment (mmagnuski) [inline mne/io/eeglab/eeglab.py]: It's actually often due to resampling in eeglab (and having boundary event at the beginning for example :) ).

File: mne/io/eeglab/tests/test_eeglab.py (hunk 2 above)
Comment (mmagnuski) [reply mne/io/eeglab/tests/test_eeglab.py]: Yes, if the value is -1 -> warning, if lower -> error.

File: mne/io/eeglab/tests/test_eeglab.py [hunk 4]
```diff
@@ -142,8 +142,11 @@ def test_io_set_raw(fnames, tmpdir):
...
+    with warnings.catch_warnings(record=True) as w:
```
Comment (mmagnuski) [inline mne/io/eeglab/tests/test_eeglab.py]: BTW, now that we are pytesting, you can pytestify this code:
```python
with pytest.warns(UserWarning, match="have a sample index of -1."):
    # ...
```

Comment (larsoner) [reply mne/io/eeglab/tests/test_eeglab.py]: In fact this pattern will be required once #5409 is in, which is almost done

File: mne/io/eeglab/eeglab.py (hunk 3 above)
Comment (jona-sassenhagen) [reply mne/io/eeglab/eeglab.py]: So should I change anything here?

**Comment by jona-sassenhagen** [comment]
Ready.

File: mne/io/eeglab/eeglab.py (hunk 3 above)
Comment (mmagnuski) [reply mne/io/eeglab/eeglab.py]: I would say something in this flavor: `""At least one event has a sample index of -1. This usually is a consequence of how eeglab handles event latency after resampling - especially when you had a boudary event at the beginning of the file. Please make sure that ..."` I'm just not sure how detailed the 'make sure part' should be. The user could just check whether the events with negative time sample were originally at the beginning of file and are boundary events or events that are not important.

**Comment by agramfort** [comment]
@mmagnuski merge if you're happy. When travis is of course :)

**Comment by mmagnuski** [comment]
@jona-sassenhagen 
Just one pep8 error:
```
mne/io/eeglab/eeglab.py:684:15: E127 continuation line over-indented for visual indent
```

**Comment by mmagnuski** [comment]
@jona-sassenhagen Codecov is complaining, maybe add a small test for the error too?

**Comment by mmagnuski** [comment]
Probably not, but I feel bounded by @agramfort allowing me to merge when CIs are green. :)
@agramfort - should I merge ignoring codecov complains?

**Comment by agramfort** [comment]
yes

thx @jona-sassenhagen

**Pull request merged**